source/common/conn_pool/conn_pool_base.cc

```diff
@@ -19,13 +19,44 @@ namespace {
 }
 } // namespace
 
+Network::TransportSocketOptionsConstSharedPtr
+wrapTransportSocketOptions(Network::TransportSocketOptionsConstSharedPtr transport_socket_options,
+                           Upstream::HostConstSharedPtr host) {
+  const auto& upstream_http_protocol_options = host->cluster().upstreamHttpProtocolOptions();
+  if (!upstream_http_protocol_options ||
+      !upstream_http_protocol_options->auto_sni_from_upstream()) {
+    return transport_socket_options;
+  }
+  const absl::string_view hostname = host->hostname();
```

Comment:
I believe `host->hostname()` can include `:port` if it was resolved by the Dynamic Forward Proxy. You can end up with invalid SNI.

Reply:
According to the comments `hostname` is a hostname and there is also `address` available in the `HostDescription`, so it would be very confusing to also include a port in `hostname`. I also can not spot where a port can be added to a hostname of a `HostImpl` or a `LogicalHost`.